crates/services/gas_price_service/src/v0/service.rs

```diff
@@ -121,24 +122,32 @@ where
     L2: L2BlockSource,
     Metadata: MetadataStorage,
 {
-    async fn run(&mut self, watcher: &mut StateWatcher) -> anyhow::Result<bool> {
-        let should_continue;
+    async fn run(&mut self, watcher: &mut StateWatcher) -> TaskRunResult {
         tokio::select! {
             biased;
             _ = watcher.while_started() => {
                 tracing::debug!("Stopping gas price service");
-                should_continue = false;
+                TaskRunResult::Stop
             }
             l2_block_res = self.l2_block_source.get_l2_block() => {
                 tracing::info!("Received L2 block result: {:?}", l2_block_res);
-                let block = l2_block_res?;
+                let block = match l2_block_res {
+                    Ok(block) => block,
+                    Err(err) => {
+                        return anyhow!(err).into()
+                    }
+                };
```

Comment:
I see this pattern in many places. Maybe it would be simpler to defined a helper trait with `into_continue_task_result` which either returns `TaskRunResult::Continue` or `TaskRunResult::ErrorContinue`